### PR TITLE
switch jinja Undefined with AnsibleUndefinedVariable

### LINKED ISSUE #63129: hostvars should use AnsibleUndefined() and not Undefined().
##### SUMMARY

In recent ansible release, the default `undefined` class in jinja evaluation context is now `AnsibleUndefined`.

This allow to chain undefined call without failing:

`foo.bar.help[0] | default(0)` will not fails if `bar` is undefined.

Unfortunately, the HostVars class used to represent the `hostvars` mapping keep returning a standard `Undefined()` when an host is not found, which make expression evaluation fails.

`hostvars.bar.help[0] | default(0)` fails.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

lib/ansible/vars/hostvars.py

##### ANSIBLE VERSION

```paste below
ansible 2.8.3
```

but the bug is still present in trunk at this moment.

##### STEPS TO REPRODUCE

```yaml
  tasks:
    - debug:
        msg: "{{ hostvars.foo.bar[0] | default('hello') }}"
```

##### EXPECTED RESULTS

print message 'hello'.

##### ACTUAL RESULTS

```
FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: \"hostvars['foo']\" is undefined…
```

### PR DESCRIPTION
**Pull request by TrayserCassa**
Fixes #63129

##### SUMMARY
I looked up at the ansible.error classes and switched it with the jinja.Undefined 

Fixes #63129

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/vars/hostvars.py

### CODE REVIEW

File: lib/ansible/vars/hostvars.py [hunk 1]
```diff
@@ -75,7 +74,7 @@ def raw_get(self, host_name):
         '''
         host = self._find_host(host_name)
         if host is None:
-            return Undefined(name="hostvars['%s']" % host_name)
+            return AnsibleUndefinedVariable(name="hostvars['%s']" % host_name)
```
Comment (mkrizek) [inline lib/ansible/vars/hostvars.py]: `AnsibleUndefinedVariable` is an exception class. You probably want `AnsibleUndefined` which is Ansible's subclass of Jinja2's `Undefined`. See https://github.com/ansible/ansible/blob/bdd0fac606265f92e6521540106accf19a1ae4d0/lib/ansible/template/__init__.py#L232-L246

Comment (TrayserCassa) [reply lib/ansible/vars/hostvars.py]: yes your right, also the error msg is better. Changed it.

**Comment by samdoran** [comment]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, please add unit tests.

Since this is changing a pretty fundamental change, please add `ci_complete` to your commit message in order to run all tests.

**Comment by mkrizek** [comment]
Superseded by https://github.com/ansible/ansible/pull/64959

**Pull request closed without merging**